src/main/java/com/glencoesoftware/pyramid/PyramidFromDirectoryWriter.java

```diff
@@ -1047,48 +1146,7 @@ else if (layoutVersion != 3) {
         LOG.debug("Merging {} original channels into {} RGB channels",
           s.c, effectiveChannels);
 
-        OMEXMLMetadataRoot root = (OMEXMLMetadataRoot) metadata.getRoot();
-        Pixels pixels = root.getImage(seriesIndex).getPixels();
-        for (int index=pixels.sizeOfChannelList()-1; index>0; index--) {
-          if (index % rgbChannels == 0) {
-            continue;
-          }
-          Channel ch = pixels.getChannel(index);
-          pixels.removeChannel(ch);
-        }
-        for (int index=0; index<pixels.sizeOfChannelList(); index++) {
-          Channel channel = pixels.getChannel(index);
-          channel.setSamplesPerPixel(new PositiveInteger(rgbChannels));
-          if (channel.getColor() != null) {
-            LOG.warn("Removing channel color");
-            channel.setColor(null);
-          }
-          if (channel.getEmissionWavelength() != null) {
-            LOG.warn("Removing channel emission wavelength");
-            channel.setEmissionWavelength(null);
-          }
-          if (channel.getExcitationWavelength() != null) {
-            LOG.warn("Removing channel excitation wavelength");
-            channel.setEmissionWavelength(null);
-          }
-          if (channel.getLightPath() != null) {
-            LOG.warn("Removing channel light path");
-            channel.setLightPath(null);
-          }
-          if (channel.getLightSourceSettings() != null) {
-            LOG.warn("Removing channel light source settings");
-            channel.setLightSourceSettings(null);
-          }
-          FilterSet filterSet = channel.getLinkedFilterSet();
-          if (filterSet != null) {
-            LOG.warn("Removing channel filter set");
-            channel.unlinkFilterSet(filterSet);
-          }
-          if (channel.getName() != null) {
-            LOG.warn("Removing channel name");
-            channel.setName(null);
-          }
-        }
+        mergeChannels(seriesIndex, rgbChannels, true);
```

Comment:
Is there a scenario where `mergeChannels` can be called twice, once with `false` above and once with `true` here?

Reply:
I think only if all of the following are true:

- `--rgb` option used
- no `OME/METADATA.ome.xml` in input Zarr
- `-f` option used with a file that has more channels than the Zarr array
- Zarr array has a multiple of 3 channels

I don't think there's a problem with calling `mergeChannels` twice in that case, but 07263e3 adds more tests.